tests/text/data/test_json_source.py

```diff
@@ -1,34 +1,59 @@
 import os
 
-from biome.text.data import DataSource
-from tests import DaskSupportTest, RESOURCES_PATH
+import pytest
+from biome.text import Dataset
+
+from tests import RESOURCES_PATH
 
 FILES_PATH = os.path.join(RESOURCES_PATH, "data")
 
 
-class JsonDatasourceTest(DaskSupportTest):
-    def test_read_json(self):
-        file_path = os.path.join(FILES_PATH, "dataset_source.jsonl")
+@pytest.fixture
+def dataset_source() -> Dataset:
+    """Creation of dataset"""
+
+    file_path = os.path.join(FILES_PATH, "dataset_source.jsonl")
+    dataset_source = Dataset.from_json(paths=file_path)
+
+    return dataset_source
+
+
+@pytest.fixture
+def dataset_flatten_source() -> Dataset:
+    """Creation of the flatten dataset"""
+
+    file_path = os.path.join(FILES_PATH, "to-be-flattened.jsonl")
+    dataset_flatten_source = Dataset.from_json(paths=file_path)
+
+    return dataset_flatten_source
+
+
+@pytest.fixture
+def dataset_nested_list() -> Dataset:
+    """Creation of the nested-list dataset"""
+
+    file_path = os.path.join(FILES_PATH, "nested-list.jsonl")
+    dataset_nested_list = Dataset.from_json(paths=file_path)
+
+    return dataset_nested_list
+
+
+def test_read_json(dataset_source):
+    """Testing JSON reading"""
 
-        datasource = DataSource(format="json", source=file_path)
-        data_frame = datasource.to_dataframe().compute()
+    assert len(dataset_source) > 0
 
-        assert len(data_frame) > 0
-        self.assertTrue("path" in data_frame.columns)
 
-    def test_flatten_json(self):
-        file_path = os.path.join(FILES_PATH, "to-be-flattened.jsonl")
-        ds = DataSource(format="json", flatten=True, source=file_path)
-        df = ds.to_dataframe().compute()
+def test_flatten_json(dataset_flatten_source):
+    """Assert that flatten operation divides correctly"""
+    dataset_flatten_source.flatten_()
 
-        for c in ["persons.*.lastName", "persons.*.name"]:
-            self.assertIn(c, df.columns, f"Expected {c} as column name")
```

Comment:
@frascuchon Do we still need this specific way of flattening `List[Dict]`s in json files (see `biome.text.data.helpers.flatten_dataframe`)? Do you remember the use case for this? If needed, we could add such a functionality directly to our `Dataset` class.

Reply:
The use case was transform datasources to parquet easily. Parquet format cannot accept automatically data object columns.

I supose we can tackle this as a `map` operation in Dataset. So, I'm ok to drop it

Reply:
:+1: